Streamline building IV

Croydon opened the pull request:
This 
  * makes various ecosystem updates to the CI infrastructure 
  * **most importantly: thanks to a lot of huge developments in the Conan ecosystem, we can finally get rid if [ci-prebuilds](https://github.com/inexorgame/ci-prebuilds) and all related logic**
    * this includes custom pre-processor logic 
    * this also includes all Conan handling logic which isn't strictly for consuming packages  
    * and a lot more....

I'm fully aware that the builds are currently broken and I'm tracking this at https://github.com/bincrafters/community/issues/604 as it is almost certainly not a problem in Inexor.

However, the current master is broken as well (see e.g. https://travis-ci.org/inexorgame/inexor-core/builds/477245258; [this particular issue is due to an old Conan version, but even if you would upgrade this you would run into the very same issue this pull requests displays]), so it makes nothing worse but makes it easier to eventually fix it (in fact, if it is really just an issue in protoc_installer as I believe it is that it will be "automatically fixed, as soon as this Conan recipe gets fixed. That Conan recipe updates are breaking retrospectively builds shouldn't happen anymore as soon as these packages are in conan-center, which is heavily work in progress for Protobuf and Protoc. It's just damn complicated... Also, Conan will soon introduce revisions as a feature which will prevent something like this in the future as well).

I'm also fully away that most people aren't interested right now to work on Core anymore due to all the elementary refactoring and redesigns going on, but I would really like to unwind ci-prebuilds sustainable.

Please merge this.

Croydon committed:
fix(appveyor): Use inexorbot instead of croydonbot

Croydon committed:
chore(appveyor): Remove cmd prefix as it is default

Croydon committed:
chore(dependencies): Update gtest to 1.8.1

Croydon committed:
chore(conanfile): Remove unneccessary comments

Croydon committed:
refactor(ci): Deleting all Conan package building + uploading

Bach when the Conan ecosystem wasn't that stable and grown it made sense that we had this centralized place to mass building all of our own Conan packages and uploading them to our own repository.

In the meantime Conan got more stable, the mass package building tools got better, the community a lot bigger and many people are focused on maintaining Conan packages.

All of these factors made this centralized, self-written mass building code needless.

Croydon committed:
refactor(ci): Update build environment

  * update Travis image to Xenial
  * update Python to 3.7
  * use newer conanio/ Docker images instead the old lasote/ ones conan-io/conan-docker-tools#58
  * normalize name of COMPILER_CONFIGURATION variable to more common name BUILD_TYPE

Croydon committed:
chore(dependencies): Switch cef package over to Bincrafters

Croydon committed:
fix(dependencies): Add protoc_installer

protoc is not any longer included in the protobuf package itself

Croydon committed:
fix(cef): Link lower case package name

Croydon committed:
chore(dependencies): Update protoc_installer to 3.6.1

Croydon committed:
chore(dependencies): Update protobuf and grpc

a-teammate commented on tool/travis.sh:
```diff
@@ -146,8 +141,8 @@ bin="${code}/bin"
-COMPILER_CONFIGURATION="${4}" # Debug or Release
-NIGHTLY="${5}" # Nightly is either true, false or conan
+BUILD_TYPE="${4}" # Debug or Release
+NIGHTLY="${5}" # Nightly is either true or false
```
maybe 
```suggestion
IS_NIGHTLY_BUILD="${5}"
```
but I am not sure from the previous comment on what premises this is set to true and when it is set to false. Could you add a short doc comment on that?

a-teammate commented:
looks much cleaner :) :+1:

Croydon committed:
chore(travis): Rename NIGHTLY to IS_NIGHTLY

Croydon committed:
chore(travis): Remove NIGHTLY_USER and NIGHTLY_PASSWORD

If nighty uploads of any kind are getting re-added in the future this can be easily added back

Croydon commented:
This can be merged now.

As I am not an administrator anymore, I can't ignore the failing status checks (reasoning explained above) and can't merge myself anymore in this case.
